### PR TITLE
Add <leader>xn and similars

### PR DESCRIPTION
**Pull request by osamuaoki**
proof-of-concept

Signed-off-by: Osamu Aoki <osamu@debian.org>

### CODE REVIEW

**Comment by osamuaoki** [comment]
This is not meant to be really merged. but as a talking point of feature request #763

**Comment by osamuaoki** [comment]
Hi,

I cleaned source a bit to make it more understandable.

If my "forced push" was a problem, I can close this and start new clean branch.
I think now it is kind-of-functional to see what I am doing

File: lua/core/mappings.lua [hunk 1]
```diff
@@ -2,14 +2,117 @@ local is_available = astronvim.is_available
...
+  else
+    if vim.g.syntax_on then
+      vim.cmd("syntax off")
+    else
+      vim.cmd("syntax on")
+    end
+    local state = bool2str(vim.g.syntax_on)
+    print(string.format("syntax %s", state))
+  end
+end
```
Comment (mehalter) [inline lua/core/mappings.lua:106]: Can you please move these to `utils.lua`

**Review by mehalter — CHANGES_REQUESTED** [review]

File: lua/core/mappings.lua [hunk 2]
```diff
@@ -2,12 +2,118 @@ local is_available = astronvim.is_available
...
+
+
 maps[""]["<Space>"] = "<Nop>"
 
 -- Normal --
 -- Standard Operations
-maps.n["<leader>w"] = { "<cmd>w<cr>", desc = "Save" }
-maps.n["<leader>q"] = { "<cmd>q<cr>", desc = "Quit" }
+-- maps.n["<leader>w"] = { "<cmd>w<cr>", desc = "Save" }
+-- maps.n["<leader>q"] = { "<cmd>q<cr>", desc = "Quit" }
```
[the remark this answers is not in the document]
Comment (osamuaoki) [reply lua/core/mappings.lua]: Yes, I did for updated branch.

File: lua/core/mappings.lua [hunk 3]
```diff
@@ -190,4 +296,13 @@ maps.t["<C-j>"] = { "<c-\\><c-n><c-w>j", desc = "Terminal down window navigation
...
 maps.t["<C-l>"] = { "<c-\\><c-n><c-w>l", desc = "Terminal right window naviation" }
 
+-- Custom menu for extra functionalities
+maps.n["xb"] = { toggle_background, desc = "Toggle background" }
+-- maps.t["xc"] = { toggle_completion, desc = "Toggle completion" }
+maps.n["xg"] = { toggle_signcolumn, desc = "Toggle signcolumn" }
+maps.n["xn"] = { change_number, desc = "Change line number" }
+maps.n["xi"] = { set_indent, desc = "Set indent" }
+maps.n["xp"] = { toggle_spell, desc = "Toggle spell" }
+maps.n["xy"] = { change_syntax, desc = "Toggle syntax" }
```
[the remark this answers is not in the document]
Comment (osamuaoki) [reply lua/core/mappings.lua]: Updated code has this feature addressed

File: lua/core/mappings.lua [hunk 4]
```diff
@@ -2,12 +2,118 @@ local is_available = astronvim.is_available
...
+    if vim.o.syntax then
+      vim.cmd("syntax off")
+    else
+      vim.cmd("syntax on")
+    end
+    local state = bool2str(vim.opt.syntax)
+    print(string.format("syntax=%s", state))
+  end
+end
+
```
[the remark this answers is not in the document]
Comment (osamuaoki) [reply lua/core/mappings.lua]: I see.  

I am start using lua/nvim after almost 1 year and 1 week of usage.  I didn't read the whole code yet.  Just trying to add feature in hackish way. 

Your guidance is highly appreciated.

File: lua/core/mappings.lua (hunk 1 above)
Comment (osamuaoki) [reply lua/core/mappings.lua]: Sure.  with astrovim

File: lua/core/mappings.lua [hunk 5]
```diff
@@ -190,4 +293,14 @@ maps.t["<C-j>"] = { "<c-\\><c-n><c-w>j", desc = "Terminal down window navigation
...
 
+-- Custom menu for extra functionalities
+maps.n["xb"] = { toggle_background, desc = "Toggle background" }
+-- maps.t["xc"] = { toggle_completion, desc = "Toggle completion" }
+maps.n["xg"] = { toggle_signcolumn, desc = "Toggle signcolumn" }
+maps.n["xn"] = { change_number, desc = "Change line numbering" }
+maps.n["xi"] = { set_indent, desc = "Change indent setting" }
+maps.n["xp"] = { toggle_spell, desc = "Toggle spellcheck" }
+maps.n["xu"] = { function() astronvim.toggle_url_match() end, desc = "Toggle URL Highlights" }
+maps.n["xy"] = { change_syntax, desc = "Toggle syntax highlight" }
```
[the remark this answers is not in the document]
Comment (osamuaoki) [reply lua/core/mappings.lua]: I suppose it worked since these are local function.  I will match style you propose.

**Comment by osamuaoki** [comment]
I will be updating https://github.com/osamuaoki/AstroNvim/tree/eXtra further.
After every thing, I think it is best to squish history.

**Comment by osamuaoki** [comment]
Since code in this branch is messy, let me close this and start new PR #769

**Pull request closed without merging**